server/util.go

```diff
@@ -55,11 +57,24 @@ func NewContext(v *viper.Viper, config *tmcfg.Config, logger log.Logger) *Contex
 // the application configuration. Command handlers can fetch the server Context
 // to get the Tendermint configuration or to get access to Viper.
 func InterceptConfigsPreRunHandler(cmd *cobra.Command) error {
-	rootViper := viper.New()
+	serverCtx := NewDefaultContext()
+
+	rootViper := serverCtx.Viper
 	rootViper.BindPFlags(cmd.Flags())
 	rootViper.BindPFlags(cmd.PersistentFlags())
 
-	serverCtx := NewDefaultContext()
+	// Get the executable name and configure the viper instance so that environmental
+	// variables are checked based off that name. The underscore character is used
+	// as a separator
+	executableName, err := os.Executable()
```

Comment:
We don't really use environment variables in the SDK, but this seems like we're introducing a standard (i.e. prefixing with the binary/executable name), which I think is a good thing, but we just need to make sure we document this somewhere.

Reply:
I'd like some tests on the order of precedence of config options parsing too (see [my comment](https://github.com/cosmos/cosmos-sdk/pull/7501#issuecomment-706542567))

Reply:
I'm refactoring some of my existing tests and working on this presently.